core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -189,7 +189,10 @@ object AclCommand extends Logging {
     }
   }
 
-  class AuthorizerService(val authorizerClassName: String, val opts: AclCommandOptions) extends AclCommandService with Logging {
+  object AuthorizerService extends Logging {
```

Comment:
If you do it like this, it changes the logger name. You probably want to have a val that sets the logger name to match the actual class versus the companion.

Reply:
Also, for long-running services, not sure this buys us much. I think this is most helpful for instances that get allocated a reasonable amount. Can we focus on those perhaps?

Reply:
For 1.  Doesn't that create a circular dependency between the class and the companion object?
For 2. Is it a good idea to leave a dangerous anti-pattern lying around in the code base for others to cut-n-paste"

Reply:
1. It does, but companions and class have a tight relationship. Is it a problem?
2. I think if we want to do more than fix the bug, then we should evaluate what is a good pattern. And I don't think the `Logging` trait is a good pattern. :)